src/main/java/org/skife/jdbi/v2/ClasspathStatementLocator.java

```diff
@@ -74,20 +75,32 @@ public static boolean looksLikeSql(String sql)
     @SuppressFBWarnings("DM_STRING_CTOR")
     public String locate(String name, StatementContext ctx)
     {
-        final String cache_key;
+        final Object cache_key;
+        boolean isSqlObjectMethod = false;
         if (ctx.getSqlObjectType() != null) {
-            cache_key = '/' + mungify(ctx.getSqlObjectType().getName() + '.' + name) + ".sql";
+            if (ctx.getSqlObjectMethod() != null) {
+                isSqlObjectMethod = true;
+                cache_key = ctx.getSqlObjectMethod();
+            } else {
+                cache_key = '/' + mungify(ctx.getSqlObjectType().getName() + '.' + name) + ".sql";
```

Comment:
If we are going to worry about object allocation here, why not do something even simpler like using a `new SimpleEntry<>(ctx.getSqlObjectType(), name)` which should be much cheaper than assembling this string?